src/scitokens_internal.h

```diff
@@ -418,24 +471,78 @@ class Validator {
     }
 
     void verify(const SciToken &scitoken, time_t expiry_time) {
-        auto result = verify_async(scitoken);
-        while (!result->m_done) {
-            auto timeout_val = result->get_timeout_val(expiry_time);
-            select(result->get_max_fd() + 1, result->get_read_fd_set(),
-                   result->get_write_fd_set(), result->get_exc_fd_set(),
-                   &timeout_val);
-            if (time(NULL) >= expiry_time) {
-                throw CurlException("Timeout when loading the OIDC metadata.");
+        std::string issuer = "";
+        auto start_time = std::chrono::steady_clock::now();
+        bool has_issuer = false;
+        
+        try {
+            // Try to extract issuer for monitoring
+            if (scitoken.m_decoded && scitoken.m_decoded->has_payload_claim("iss")) {
```

Comment:
It's not clear to me the decoding step always occurs first as is assumed here - why do you think this is a useful code path?